Modules/_ssl/cert.c

```diff
@@ -191,19 +195,20 @@ certificate_hash(PySSLCertificate *self)
 }
 
 static PyObject *
-certificate_richcompare(PySSLCertificate *self, PyObject *other, int op)
+certificate_richcompare(PyObject *lhs, PyObject *rhs, int op)
 {
     int cmp;
+    PySSLCertificate *self = _PySSLCertificate_CAST(lhs);
     _sslmodulestate *state = get_state_cert(self);
 
-    if (Py_TYPE(other) != state->PySSLCertificate_Type) {
+    if (Py_TYPE(rhs) != state->PySSLCertificate_Type) {
         Py_RETURN_NOTIMPLEMENTED;
     }
     /* only support == and != */
     if ((op != Py_EQ) && (op != Py_NE)) {
         Py_RETURN_NOTIMPLEMENTED;
     }
-    cmp = X509_cmp(self->cert, ((PySSLCertificate*)other)->cert);
+    cmp = X509_cmp(self->cert, ((PySSLCertificate*)rhs)->cert);
```

Comment:
```suggestion
    cmp = X509_cmp(self->cert, _PySSLCertificate_CAST(rhs)->cert);
```

Reply:
I've left out fast casts for which the input has already been checked so that if the macro is one day converted to a function with explicit checks, we don't double checks those